Ch/dev tm

cleherny opened the pull request:
## What?
Developments of scripts for quarry tracking over different datasets and evolution monitoring. Take into account the development relative to RDP algorithm. The project and requirements, scripts, data and steps to run the project are described in detail. A shell script is provided to automatized prediction workflow.
## Why?
The object tracking tool was the main improvement to complete the quarry project. Adding detailed information was also required for documentation and in order the user should be able to run the code on his own following instructions. The implementation of the RDP algorithm in the object-detector project allow to reduced the prediction file size which is important for proj-dqry as a lot of detection are stored.
## How?
Writing of new scripts and config files to track objects with the script detection_monitoring.py and plot some results with plot.py. A new config-dm.yaml file allow to set the input parameters. README.md files have been written and can be found in main (presentation of the project and requirements), scripts (described the scripts actions) and config (data and instruction to run the project) folders. A shell script (batch_process.sh) allow to execute commands to perform quarry prediction for several years and works along with a config template config-prd.template.yaml adapted for a given year.

cleherny commented:
The PR#5 is ready for review. In particular the README.md must be checked. Scripts have already been reviewed.

GwenaelleSa reviewed (approved):
I reviewed the two readme of the project. I tested the commands for the training. For the detection part, I could not download all the images. I tested for 2019 and 2015. I ran the script for the tileset generation several times, but it never worked.

The readme are globally clear for the main one, I made the modifications directly in a local file: 
[README (1).md](https://github.com/swiss-territorial-data-lab/proj-dqry/files/13954002/README.1.md). For the one about the scripts, I made comments in the review.

I have several feedbacks about the image of the main readme:
- geodata in one word
- "Reviewed labels" with uppercase at "labels", like in "Acquisition footprint".
- Why put "(iteration)" in the optimum model choice?
- For the training you wrote "detections" and for the inference "inference detections". Was it wanted?
- "Detection filtering" without "s" to "detection" as it is a compound word.

In the same readme, the part between parenthesis at line 251 is unclear to me: "(filtered detection files (`oth_detections_filter_year-{year}_[filters_list].geojson`))". Maybe you could precise that this file is an output of the detection workflow. Also, is the {year} a parameter to change or is it dealt with by the scripts?

For the config files, the name is not really intuitive for `config_dt.yaml`. Especially, "dt" is very close from "det" and I don't think the acronym is explained somewhere.

Is it wanted that most of the time, "detection" is singular?

cleherny commented on README.md:
Deleted

cleherny commented on scripts/README.md:
```diff
@@ -0,0 +1,194 @@
+## Overview
+
+Following, the scripts contained in `proj-dqry` are presented in detailed. They are used to pre- and post-process data and results as part of the automatic detection of mineral extraction sites (MES, also referred as quarry in this project) automatic detection.
```
I replaced the paragraph by:
"The following is a detailed description of the scripts contained in `proj-dqry`. They are used to process data and results as part of the automatic detection of mining extraction sites (MES, also known as quarries in this project) with the STDL's object detector."

cleherny commented on README.md:
I added commands in README

cleherny commented on scripts/README.md:
````diff
@@ -0,0 +1,194 @@
+    output: ./output/output_det_/oth_detection_at_0dot*_threshold_year-[YEAR]_score-{score}_area-{area}_elevation-{elevation}_distance-{distance}.geojson
+```
+
+-**year**: year of the dataset used as input for filtering
````
I homogenised the description of elements. Valid for all the other similar comments.

cleherny commented on scripts/README.md:
```diff
@@ -0,0 +1,194 @@
+
+-**input**: indicates the path to the input file that needs to be filtered, _i.e._ `oth_detections_at_0dot*_threshold.gpkg`
+
+-**labels_shapefile**: AoI of interest is used to remove polygons located partially outside the AoI. _SWISSIMAGE_ acquisition footprint for a given year (swissimage_footprint_[YEAR].shp) were used for this project.
```
I changed it to "shapefile"

cleherny commented on scripts/README.md:
````diff
@@ -0,0 +1,194 @@
+$ python  <dir_path>/scripts/detections_tracking.py <dir_path>/config/config_dt.yaml
+```
+
+The outputs are a _geojson_ and _csv_ (**quarry_time**) files saving detections over the years with their characteristics (ID_object, ID_feature, year, score, area, geometry). 
````
It's the name of the results. I changed it to .geojson and .csv to "detections_years".

cleherny commented on scripts/README.md:
```diff
@@ -0,0 +1,194 @@
+
+The script `detections_tracking.py` has been developed to identify and track an object between different year datasets. It works along with the config file `config_dt.yaml`.
+
+The script identifies the position of polygons between different years to identify a single object in multiple datasets. All detected shapes are added to a single data frame. A union function is applied to merge overlapping polygons. Those polygons (single and merged) are added to a new data frame with a unique ID attributed to each. This data frame is then spatially compared to the initial data  ...
```
"The detection shapefiles for each year are merged. Overlapping groups of detections and non-overlapping individual detections are assigned a unique object identifier. The object identifier is then propagated to each year's detection polygons by intersection with the merged polygons. Polygons sharing the object identifier are assumed to describe the same object, allowing it to be tracked over time."

cleherny commented:
Thanks for your review. I updated your comments and did additional changes.

- On my side, there is no issue to donwload tiles (except for the 2020 problematic ones). Did you assign a year to the url in `config_det.yaml`? I added a new sentence in the main README to warn people.
- I changed `config_dt.yaml` to `config_track.yaml`. dt stands for "detection tracking".
- I use 'detection' when I refer to the action and 'detections' when I refer to the objects.

GwenaelleSa commented on scripts/README.md:
````diff
@@ -0,0 +1,194 @@
+$ python <dir_path>/scripts/filter_detections.py <dir_path>/config/config_det.yaml
+```
+
+## Detections tracking
````
But then it is the only title starting with an infinitive verb.

cleherny replied:
No, it isn't the only one: "Prepare data", "Filter detections", "Get DEM". I see no issue with not having strictly the same title "style".
